### ISSUE TITLE
[Possible Issue] Accessing a falsy value in lib/fetcher

### ISSUE THREAD

**Issue by Exhabition** [Butterstroke/AniList-Node#28]
**Description**
In `lib/fetcher` I found a possible issue. Read the following code.

https://github.com/Butterstroke/AniList-Node/blob/68c8006331ac304fbc11eddb4e64f4b4bccf08dd/lib/fetcher.js#L129-L137

The first statement returns errors if there are any, this means that if there are no errors it will continue to the next statement.
We now know that `json.errors` is a falsy value because otherwise it would have already returned.

However, in the second statement, `if (json.date === null)`, we are accessing `json.errors` anyway.

TL;DR: Accesing `json.errors` after we now it's a falsy value

**Environment**
AniList-Node Version: 1.8.1

**Additional context**
Code in: `lib/fetcher`, lines 129 - 137

**Comment by Butterstroke** [comment]
Looks like this is a stray line of code that was meant to be removed in #24. Error handling for data changed recently within AniList-Node to handle 401 errors since they were not caught before. This should not effect anything since if `json.data === null`, there is always a `json.errors` that is 404. 

Nothing too serious but lines 135-137 should get removed at some point because it does nothing.

**Comment by Exhabition** [comment]
I see, just wanted to point that out, since I was looking around in `lib/fetcher` some more, a question popped up.

https://github.com/Butterstroke/AniList-Node/blob/68c8006331ac304fbc11eddb4e64f4b4bccf08dd/lib/fetcher.js#L71-L73
https://github.com/Butterstroke/AniList-Node/blob/68c8006331ac304fbc11eddb4e64f4b4bccf08dd/lib/fetcher.js#L28-L50

Is there any point in using await on `edgeRemove`? It doesn't return a promise. If the awaiting of those functions is removed, function `formatMedia` for example doesn't have to be executed asynchronous anymore.

Unless there is a thought behind this, in that case I would like to know the explanation 😄

**Comment by Butterstroke** [comment]
> https://github.com/Butterstroke/AniList-Node/blob/68c8006331ac304fbc11eddb4e64f4b4bccf08dd/lib/fetcher.js#L28-L50
> 
> Is there any point in using await on `edgeRemove`? It doesn't return a promise. If the awaiting of those functions is removed, function `formatMedia` for example doesn't have to be executed asynchronous anymore.
> 
> Unless there is a thought behind this, in that case I would like to know the explanation 😄

Hmm... I believe the thought was to ensure the object was edited before it was returned. I'm trying to think if edgeRemove was always synchronous or not but I can't say for sure since it's been around forever. Might be worth a look to see if synchronous is just as good for it.

**Comment by Butterstroke** [comment]
Made a new ticket for the async/sync discussion so that I can close this issue. The stray lines at 135-137 have been removed in the development branch.
